pkg/apis/batch/types.go

```diff
@@ -256,6 +256,50 @@ type PodFailurePolicy struct {
 	Rules []PodFailurePolicyRule
 }
 
+// SuccessPolicy describes when a Job can be declared as succeeded based on the success of some indexes.
+type SuccessPolicy struct {
+	// Criteria represents the list of alternative criteria for the declaring the Jobs
+	// as successful before `.status.succeeded >= .spec.completions`. Once any of the criteria are met,
+	// the "SucceededCriteriaMet" condition is added, and the lingering pods are removed.
+	// The terminal state for such a Job has the "Completed" condition.
+	// Additionally, these criteria are evaluated in order; Once the Job meets one of the criteria,
+	// other criteria are ignored.
+	// +listType=atomic
+	Criteria []SuccessPolicyCriteria
```

Comment:
I would advise `rules` to match `PodFailurePolicy`. But up to @deads2k

Reply:
rules is a good idea.

Reply:
@alculquicondor @deads2k If so, do we ok implementations changes, and after we update KEP, right?

In the case of replace with `Rules`, I think that we need to replace `SuccessCriteriaMet` with `SuccessTarget`, right?

Reply:
cc: @soltysh

Reply:
makes sense from a consistency perspective, although `SuccessCriteriaMet` conveys the meaning  better IMO

Reply:
> makes sense from a consistency perspective, although SuccessCriteriaMet conveys the meaning better IMO

Yea, I think so too, but I believe that consistency is more important thing perspective of UX.

Reply:
> If so, are we ok implementations changes first, and after we update KEP, right?

Yes

Keep `SuccessCriteriaMet`

Reply:
Based on this discussion, I'll try to replace only the `SuccessPolicyCriteria` (`JSON=criteria`) with `SuccessPolicyRules` (`JSON=rules`).

It means that we keep using `SuccessCriteriaMet`. @soltysh @deads2k @alculquicondor @mimowo @atiratree If anyone prefers to replace `SuccessCriteriMet` with `SuccessTarget`, let me know.

Reply:
I'm fine with SuccessPolicyMet, SuccessCriteriaMet and SuccessTarget, no particular preference, all have their merit imo.

Reply:
Thanks for sharing your opinion.

Reply:
I replaced the `SuccessPolicyCriteria` with `SuccessPolicyRule` here: https://github.com/kubernetes/kubernetes/pull/123412/commits/18795c121cf0ade6c82be76ebf6cbfb0346ab032

Reply:
I am in favor of SuccessCriteriaMet or SuccessPolicyMet.